plugins/module_utils/ilo_redfish_utils.py

```diff
@@ -51,6 +99,423 @@ def get_ilo_sessions(self):
         result["ret"] = True
         return result
 
+    def get_systems_data(self):
+        response = self.get_request(self.root_uri + self.systems_uri)
+        if not response["ret"]:
+            return response
+
+        return {
+            "ret": True,
+            "msg": response
+        }
+
+    def get_managers_data(self):
+        response = self.get_request(self.root_uri + self.manager_uri)
+        if not response["ret"]:
+            return response
+
+        return {
+            "ret": True,
+            "msg": response
+        }
+
+    def get_service_bios_attributes(self):
+        # This method returns service BIOS attributes
+        response = self.get_systems_data()
+        if not response["ret"]:
+            return response
+
+        data = response["msg"]["data"]
+
+        if "Bios" not in data:
+            return {
+                "ret": False,
+                "msg": "Getting BIOS URI Failed, Key 'Bios' "
+                       "not found in %s response: %s"
+                       % (self.root_uri + self.service_root, str(data))
+            }
+
+        bios_uri = data["Bios"]["@odata.id"]
+
+        # Get service settings
+        service_uri = bios_uri + "service/settings/"
+        response = self.get_request(self.root_uri + service_uri)
+        # Check if service API doesn't support
+        if not response["ret"]:
```

Comment:
Looking at this logic a bit more, it seems to only go into the OEM path if the standard path failed. Is this a realistic scenario?

Reply:
Nevermind; `bios_uri + "service/settings/"` isn't a standard URI path, but it also doesn't contain `Oem` like the Redfish spec defines for OEM URIs. What exactly is found at `bios_uri + "service/settings/"`?

Reply:
bios_uri + "service/settings/" - is for Gen10 server model
bios_uri + "oem/hpe/service/settings/" - is for Gen10 Plus server model 
Both will fetch the service BIOS attributes.

Reply:
Removed this functionality for now. Resolving this conversation, thanks!